extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java

```diff
@@ -118,28 +134,49 @@ public void accept(Throwable throwable) {
             event.response().end();
             return;
         }
-        String accept = event.request().getHeader("Accept");
-        if (accept != null && accept.contains("application/json")) {
-            event.response().headers().set(HttpHeaderNames.CONTENT_TYPE, "application/json; charset=utf-8");
-            String escapedDetails = escapeJsonString(details);
-            String escapedStack = escapeJsonString(stack);
-            StringBuilder jsonPayload = new StringBuilder("{\"details\":\"")
-                    .append(escapedDetails)
-                    .append("\",\"stack\":\"")
-                    .append(escapedStack)
-                    .append("\"}");
-            writeResponse(event, jsonPayload.toString());
-        } else {
-            //We default to HTML representation
-            event.response().headers().set(HttpHeaderNames.CONTENT_TYPE, "text/html; charset=utf-8");
-            final TemplateHtmlBuilder htmlBuilder = new TemplateHtmlBuilder("Internal Server Error", details, details);
-            if (showStack && exception != null) {
-                htmlBuilder.stack(exception);
-            }
-            writeResponse(event, htmlBuilder.toString());
+
+        if (responseContentType == null) {
+            responseContentType = "";
+        }
+        switch (responseContentType) {
+            case ContentTypes.TEXT_HTML:
+            case ContentTypes.APPLICATION_XHTML:
+            case ContentTypes.APPLICATION_XML:
+            case ContentTypes.TEXT_XML:
+                htmlResponse(event, details, exception);
+                break;
+            case ContentTypes.APPLICATION_JSON:
+            case ContentTypes.TEXT_JSON:
+                jsonResponse(event, responseContentType, details, stack);
+                break;
+            // We default to JSON representation
+            default:
+                jsonResponse(event, ContentTypes.APPLICATION_JSON, details, stack);
+                break;
         }
     }
 
+    private void jsonResponse(RoutingContext event, String contentType, String details, String stack) {
+        event.response().headers().set(HttpHeaderNames.CONTENT_TYPE, contentType + "; charset=utf-8");
+        String escapedDetails = escapeJsonString(details);
+        String escapedStack = escapeJsonString(stack);
+        StringBuilder jsonPayload = new StringBuilder("{\"details\":\"")
```

Comment:
Unrelated to this PR - I'm wondering if at some point we shouldn't use something like https://github.com/zalando/problem.